.circleci/config.yml

```diff
@@ -1,237 +1,197 @@
-  working_directory: ~/go/singularity
-  machine:
-    image: ubuntu-2004:202008-01
-
-x-run:
-  setup_environment: &setup_environment
-    name: Setup environment
-    command: |-
-      echo 'set -x' >> $BASH_ENV
-      echo 'export GOPATH=$HOME/go'           >> $BASH_ENV
-      echo 'export GOROOT=/usr/local/go' >> $BASH_ENV
-      echo 'export GOBIN=$HOME/go/bin'        >> $BASH_ENV
-      echo 'export PATH=$GOBIN:$GOROOT/bin:/bin:/sbin:/usr/bin:/usr/sbin:/usr/local/bin:/usr/local/sbin' >> $BASH_ENV
-      env | sed -e 's,^,ENV: ,' | sort
-      test -e $BASH_ENV && sed -e 's,^,BASH_ENV: ,' < $BASH_ENV
-
-  update_go: &update_go
-    name: Update Go to 1.14.9
-    working_directory: /tmp
-    command: |-
-      wget https://dl.google.com/go/go1.14.9.linux-amd64.tar.gz
-      sudo rm -rf $GOROOT
-      sudo tar -C /usr/local -xzf go1.14.9.linux-amd64.tar.gz
-      sudo ln -s $GOROOT/bin/go /usr/local/bin/go
-
-  fetch_deb_deps: &fetch_deb_deps
-    name: Fetch deps
-    working_directory: /tmp
-    command: |-
-      # https://discuss.circleci.com/t/could-not-get-lock-var-lib-apt-lists-lock/28337/4
-      sudo killall -9 apt-get || true && \
-      sudo apt-get update -y && \
-      sudo apt-get install -y build-essential squashfs-tools libseccomp-dev cryptsetup
-
-  buildtest_rpm: &buildtest_rpm
-    name: Build and test rpm under docker
-    no_output_timeout: 10m
-    command: |-
-      export E2E_DOCKER_USERNAME=$CIRCLE_CI_DOCKER_USERNAME && \
-      export E2E_DOCKER_PASSWORD=$CIRCLE_CI_DOCKER_PASSWORD && \
-      scripts/ci-docker-run
-
-  build_singularity: &build_singularity
-    name: Build Singularity
-    command: |-
-      ./mconfig -v -p /usr/local
-      make -C ./builddir all
-
-  check_singularity: &check_singularity
-    name: Check Singularity
-    command: |-
-      make -C ./builddir check
-
-  install_singularity: &install_singularity
-    name: Install Singularity
-    command: |-
-      sudo make -C ./builddir install
-
-  check_changes: &check_changes
-    name: Check changes
-    command: |-
-      rc=0
-      scripts/should-e2e-run "${CIRCLE_PROJECT_USERNAME}" "${CIRCLE_PROJECT_REPONAME}" "${CIRCLE_BRANCH}" "${CIRCLE_PULL_REQUEST}" || rc=$?
-      case $rc in
-        0)
-          echo "Verifying critical changes"
-          ;;
-
-        1)
-          echo "No critical changes, skipping tests"
-          circleci step halt
-          ;;
-
-        *)
-          echo "E: scripts/should-e2e-run returned with exit code $rc. Abort."
-          exit $rc
-          ;;
-      esac
-
-jobs:
-  check_go_mod:
+orbs:
+  go: circleci/go@1.6
+
+parameters:
+  go-version:
+    type: string
+    default: 1.14.9
+
+executors:
+  golang:
+    parameters:
+      variant:
+        description: "variant (ex. alpine)"
+        default: ""
+        type: string
     docker:
-      - image: golang:1.14
+      - image: golang:<< pipeline.parameters.go-version >><<# parameters.variant >>-<< parameters.variant >><</ parameters.variant >>
```

Comment:
One small request... can you put a comment here giving example(s) of what this can resolve to?

The syntax is more than a bit unusual to people unfamiliar (like me :smile:)